module/VuFind/src/VuFind/ILS/Driver/AbstractBase.php

```diff
@@ -66,6 +70,170 @@ public function setConfig($config)
+     * Each value is trimmed if they are not null.
+     *
+     * @param ?string                        $firstname        Profile first name
+     * @param ?string                        $lastname         Profile last name
+     * @param string                         $birthdate        Y-m-d or an empty string
+     * @param ?string                        $address1         Address 1
+     * @param ?string                        $address2         Address 2
+     * @param ?string                        $city             City
+     * @param ?string                        $country          Country
+     * @param ?string                        $zip              Postal code
+     * @param ?string                        $phone            Phone number
+     * @param ?string                        $mobile_phone     Mobile phone number
+     * @param ?string                        $expiration_date  Profile expiration date
+     * @param TranslatableString|string|null $group            Group i.e. Student, Staff, Faculty, etc
+     * @param ?string                        $home_library     The locationID value of a pick-up location
+     *                                                         (see getPickUpLocations)
+     *                                                         that should be used as the patron's default
+     * @param array                          $nonDefaultFields Non default fields not documented in the documentation.
+     *                                                         Merges into the resulting profile array.
+     *
+     * @see https://vufind.org/wiki/development:plugins:ils_drivers#getmyprofile
+     *
+     * @return array
+     */
+    public function createProfileArray(
+        ?string $firstname = null,
+        ?string $lastname = null,
+        string $birthdate = '',
+        ?string $address1 = null,
+        ?string $address2 = null,
+        ?string $city = null,
+        ?string $country = null,
+        ?string $zip = null,
+        ?string $phone = null,
+        ?string $mobile_phone = null,
+        ?string $expiration_date = null,
+        TranslatableString|string|null $group = null,
+        ?string $home_library = null,
+        array $nonDefaultFields = []
+    ): array {
+        $profile = compact(
+            'firstname',
+            'lastname',
+            'birthdate',
+            'address1',
+            'address2',
+            'city',
+            'country',
+            'zip',
+            'phone',
+            'mobile_phone',
+            'expiration_date',
+            'group',
+            'home_library'
+        );
+        $this->debugDriverResult(__FUNCTION__, $profile, $nonDefaultFields);
+
+        if ($nonDefaultFields) {
+            $profile = array_merge($profile, $nonDefaultFields);
+        }
+        return array_map(
+            function ($value) {
+                if ($value === null || $value instanceof TranslatableString) {
```

Comment:
If you change the above, you'll also want to change this (and make sure the corresponding use statement is present):

```suggestion
                if ($value === null || $value instanceof Stringable) {
```